languages/python/reference/README.md

```diff
@@ -1,5 +1,598 @@
+- [PEP 8](./concepts/pep_8_style_guide.md) style conventions
+
+## Object-oriented concepts
+
+- [Classes](../../../reference/concepts/classes.md)
+- [Composition](../../../reference/concepts/composition.md)
+- [Encapsulation](../../../reference/concepts/encapsulation.md)
+- [Inheritance](../../../reference/concepts/inheritance.md)
+- [Interfaces](../../../reference/concepts/interfaces.md)
+- [Mutation](../../../reference/concepts/mutation.md)
+- [Objects](../../../reference/concepts/objects.md)
+- [Polymorphism](../../../reference/concepts/polymorphism.md)
+- [State](../../../reference/concepts/state.md)
+
+## Functional concepts
+
+- [Anonymous functions](../../../reference/concepts/anonymous_functions.md) (Python's `lambda`s)
+- [Higher-order functions](../../../reference/concepts/higher_order_functions.md)
+- [Immutability](../../../reference/concepts/immutability.md) (of certain builtin primitives)
+- [Nested functions](../../../reference/concepts/nested_functions.md)
+- [Partial application](../../../reference/concepts/partial_application.md)
+- [Pipelines](../../../reference/concepts/pipelines.md)
+- [Pure functions](../../../reference/concepts/pure_functions.md)
+- [Recursion](../../../reference/concepts/recursion.md)
+- [REPL](../../../reference/concepts/repl.md)
+- [Type inference](../../../reference/concepts/type_inference.md)
+
+## General concepts
+
+- [Arithmetic](../../../reference/concepts/arithmetic.md)
+- [Bitwise manipulation](../../../reference/concepts/bitwise_manipulation.md)
+- [Boolean logic](../../../reference/concepts/boolean_logic.md)
+- [Comments](../../../reference/concepts/comments.md)
+- [Conditionals](../../../reference/concepts/conditionals.md)
+- [Enumeration](../../../reference/concepts/enumeration.md)
+- [Functions](../../../reference/concepts/functions.md)
+- [Generics](../../../reference/concepts/generics.md)
+- [Locking](../../../reference/concepts/locking.md)
+- [Loops](../../../reference/concepts/loops.md)
+- [Methods](../../../reference/concepts/methods.md)
+- [Scope](../../../reference/concepts/scope.md)
+- [Variables](../../../reference/concepts/variables.md)
+
+## Types
+
+- [array](../../../reference/types/array.md)
+- [bool](../../../reference/types/boolean.md)
+- [bytes](../../../reference/types/bytes.md)
+- [class](../../../reference/types/class.md)
+- [float](../../../reference/types/floating_point_number.md)
+- [decimal](../../../reference/types/decimal_number.md)
+- [deque](../../../reference/types/deque.md)
+- [dict](../../../reference/types/hash_map.md)
+- [int](../../../reference/types/integer.md)
+- [list](../../../reference/types/list.md)
+- [None](../../../reference/types/null.md)
+- [set](../../../reference/types/set.md)
+- [str](../../../reference/types/string.md)
+- [struct](../../../reference/types/struct.md)
+- [tuple](../../../reference/types/tuple.md)
+
+## Resources used
+
+- https://www.python.org
+
+## Extracted Concepts
+
+### [Dictionary][dictionary]
```

Comment:
The headers don't point to any resource. Was that intentional?

Reply:
Yeah, for now ... my plan is for all of these to be moved into discrete files, either they're general concepts that need to referenced in [REPO]/reference/concepts or they're truly Python-specific and need to be referenced in [LANGUAGE]/reference/concepts.  We'll do the work of divying them up from the merged master.

Reply:
> my plan is for all of these to be moved into discrete files

Great idea!